torchao/experimental/kernels/cpu/aarch64/tests/CMakeLists.txt

```diff
@@ -35,13 +40,29 @@ endif()
 
 add_subdirectory(${TORCHAO_LIBRARIES}/torchao/experimental/kernels/cpu/aarch64 ${CMAKE_CURRENT_BINARY_DIR}/torchao_kernels_aarch64)
 
-# The TORCHAO_ENABLE_KLEIDI cmake variable should be set by `torchao_kernels_aarch64"
-if(TORCHAO_ENABLE_KLEIDI)
+# The TORCHAO_BUILD_KLEIDI cmake variable should be set by `torchao_kernels_aarch64"
+if(TORCHAO_BUILD_KLEIDI)
   add_compile_definitions(TORCHAO_ENABLE_KLEIDI)
 endif()
 
+if(TORCHAO_BUILD_ARM_I8MM)
+  add_compile_definitions(TORCHAO_ENABLE_ARM_I8MM)
+endif()
+
 enable_testing()
 
+if (ANDROID_ABI)
+  # Given where we are today this is sufficent. But needs to be revisited.
+  # This is also needed for native builds, but keeping it only for cross builds
+  # for now given the hacky nature.
+  file(GLOB DOTPROD_SRC_FILES test*.cpp)
```

Comment:
Are you clubbing dotprod related tests in the same PR? I thought this PR was about i8mm

Reply:
so test*.cpp files don't contain kernels, but kernels are header-only. So flag needs to pass here.

Also the test*.cpp pulls two different things dotprod header kernels and kleidi i8mm kernels. Kleidi modules get i8mm flag from Kleidi CMake but here our headers don't get dotprod. It wasn't necessary until now because, i am guessing, on m1 clang that's enabled by default. That is not the case for NDK clang.